nemoguardrails/llm/providers/providers.py

```diff
@@ -228,6 +228,20 @@ def get_llm_provider(model_config: Model) -> Type[BaseLanguageModel]:
                 "Could not import langchain_openai, please install it with "
                 "`pip install langchain-openai`."
             )
+
+    elif model_config.engine == "vertexai":
```

Comment:
This is tricky, it solves the warning for langchain <0.2.0, but if in any future langchain version they decide the provider for VertexAI will be a different class we will still use the one for 0.2.0. 

On my side, it's better not to manually set providers as langchain does that automatically for their library and they maintain this better than us.

Reply:
I think this solution is ok. This is mainly due to this intermediate phase where LangChain is migrating from providing the LLM providers from the main library vs. individual packages. The former is being deprecated. In the future, the import from the main library will not be available anyway.